StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs

```diff
@@ -963,5 +963,54 @@ private void Test2(string description = null, Func<object, string> resolve = nul
 
             await VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
         }
+
+        [Fact]
+        [WorkItem(3510, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3510")]
+        public async Task TestDelegateUsedInCastAsync()
+        {
+            var testCode = @"
+using System;
+
+public class TypeName
+{
+    public void Test()
+    {
+        var z1 = (Func<int>){|#0:delegate|} { return 1; };
+        var z2 = (Func<int>){|#1:delegate|}() { return 1; };
+        var z3 = (Func<int, int>){|#2:delegate|} { return 1; };
+        var z4 = (Func<int, int>){|#3:delegate|}(int x) { return 1; };
+        var z5 = (Func<int, int, int>){|#4:delegate|} { return 1; };
+        var z6 = (Func<int, int, int>){|#5:delegate|}(int x, int y) { return 1; };
```

Comment:
💡 These can use an even simpler syntax since none of the items in `expected` have any properties set aside from location.

```suggestion
        var z1 = (Func<int>)[|delegate|] { return 1; };
        var z2 = (Func<int>)[|delegate|]() { return 1; };
        var z3 = (Func<int, int>)[|delegate|] { return 1; };
        var z4 = (Func<int, int>)[|delegate|](int x) { return 1; };
        var z5 = (Func<int, int, int>)[|delegate|] { return 1; };
        var z6 = (Func<int, int, int>)[|delegate|](int x, int y) { return 1; };
```